### PR TITLE
IWB-2: added blacklist management functionality

### PR DESCRIPTION
**Pull request by FrankreedX**
This is a crude effort. I'm aware of the lack of polish.
Some features that will be added in a future PR:

- Loading spinner
- Bulk removal (probably will replace the current unban buttons) + big unban button at the bottom
- Snackbar

### CODE REVIEW

File: app/content/team/index.js [hunk 1]
```diff
@@ -252,48 +252,62 @@ function Index() {
...
-                    anchorPosition="bottom"
-                    contentStyle={{
-                      backgroundColor: themeColors.background,
-                    }}
-                  >
-                    <Menu.Item
-                      leadingIcon="pencil-outline"
+                <>
+                  {currentUserInfo.role === "coach" ||
+                  currentUserInfo.role === "owner" ? (
```
Comment (Gehrkej) [inline app/content/team/index.js:257]: In the case where we don't add anymore roles, could we do this?
```suggestion
                  {currentUserInfo.role !== "player" ? (
```
Might be better to leave it like this though. Just a thought

File: app/segments/(team)/manageForeignRequests.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  };
+
+  const tabComponent = useMemo(
+    () => ({
+      invites: <></>,
+      waitlist: <></>,
+      blacklist: <Blacklist />,
+    }),
+    [],
+  ); // Recreate pages only if drillId changes
```
Comment (Gehrkej) [inline app/segments/(team)/manageForeignRequests.js:45]: Can you explain this comment a bit?

File: app/segments/(team)/blacklist.js
Comment (Gehrkej) [inline app/segments/(team)/blacklist.js:1]: Unsure of what is causing this right now but some of the emails are blank: 
image

File: dbOperations/removeBlacklist.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { deleteDoc, doc } from "firebase/firestore";
+import { db } from "~/firebaseConfig";
+
+async function removeBlacklist(teamId, userId) {
+  try {
+    await deleteDoc(doc(db, "teams", teamId, "blacklist", userId));
+  } catch (e) {
+    console.log("Remove Blacklist Transaction failed: ", e);
```
Comment (Gehrkej) [inline dbOperations/removeBlacklist.js:8]: should this say transaction? 
```suggestion
    console.log("Remove User from Blacklist failed: ", e);
```

File: update_unique.py
Comment (Gehrkej) [inline update_unique.py:1]: Is this script to add emails to the blacklist entries? we will probably want to remove any scripts from version control when we start making builds or at least ignore them.

File: app/segments/(team)/blacklist.js
Comment (Gehrkej) [reply app/segments/(team)/blacklist.js:1]: This might be because I remove users after test, I will try to see who is missing from the list

**Review by Gehrkej — CHANGES_REQUESTED** [review]
I really like the approach you took with the user management stuff! Just a couple small changes and tracking down a UI bug that might've been caused from testing

File: app/segments/(team)/blacklist.js
Comment (Gehrkej) [reply app/segments/(team)/blacklist.js:1]: It looks like the new entries made using the current blacklist functionality lack an email, probably is why you had the script 
image

Comment (Gehrkej) [reply app/segments/(team)/blacklist.js:1]: I have created an issue to address this https://github.com/Golf-Drill-Challenge-App/Golf-App/issues/301

File: app/content/team/index.js (hunk 1 above)
Comment (FrankreedX) [reply app/content/team/index.js:257]: I'd say let's keep it explicit

File: app/segments/(team)/manageForeignRequests.js (hunk 2 above)
Comment (FrankreedX) [reply app/segments/(team)/manageForeignRequests.js:45]: I copied this from content/drill/[id] so it's a leftover comment

File: update_unique.py
Comment (FrankreedX) [reply update_unique.py:1]: This script was used to add the "email" field to each blacklist entry, and ye you're right. I'll prob remove it in a future PR. Although I can't deny the usefulness of having these old scripts around to reference and change when I need to do sth

File: app/segments/(team)/blacklist.js
Comment (FrankreedX) [reply app/segments/(team)/blacklist.js:1]: Not sure if this is an "issue" because it'll be resolved once this PR is merged.

File: update_unique.py
Comment (Gehrkej) [reply update_unique.py:1]: They are super useful for sure, we could have them be ignored in the build or we could make a new repo with just these scripts 🤷‍♂️

File: app/segments/(team)/blacklist.js
Comment (Gehrkej) [reply app/segments/(team)/blacklist.js:1]: I addressed it in #302

**Pull request merged**